controller/src/main/java/io/pravega/controller/server/rest/RESTServer.java

```diff
@@ -68,12 +74,37 @@ protected void startUp() {
         long traceId = LoggerHelpers.traceEnterWithContext(log, this.objectId, "startUp");
         try {
             log.info("Starting REST server listening on port: {}", this.restServerConfig.getPort());
-            httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            if (restServerConfig.isTlsEnabled()) {
+                SSLContextConfigurator sslCon = new SSLContextConfigurator();
+                sslCon.setKeyStoreFile(restServerConfig.getKeyFilePath());
+                sslCon.setKeyStorePass(loadPasswordFromFile(restServerConfig.getKeyFilePasswordPath()));
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true,
+                        new SSLEngineConfigurator(sslCon, false, false, false));
+            } else {
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            }
         } finally {
             LoggerHelpers.traceLeave(log, this.objectId, "startUp", traceId);
         }
     }
 
+    private String loadPasswordFromFile(String keyFilePasswordPath) {
+
+        if (Strings.isNullOrEmpty(keyFilePasswordPath)) {
+            return "";
+        }
+        File passwdFile = new File(keyFilePasswordPath);
+        if (passwdFile.length() == 0) {
+            return "";
```

Comment:
Same question here about returning an empty string.

Reply:
The comment in the code covers this path too.